lib/ansible/plugins/action/yum.py

```diff
@@ -0,0 +1,91 @@
+# (c) 2018, Ansible Project
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.plugins.action import ActionBase
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+class ActionModule(ActionBase):
+
+    TRANSFERS_FILES = False
+
+    def run(self, tmp=None, task_vars=None):
+        '''
+        Action plugin handler for yum3 vs yum4(dnf) operations
+
+        In the past, there was only yum which is now known as yum3 or yum 3.x
+        but now there is a "new yum" known as yum4 or yum 4.x which is actually
+        just a yum command-line compatibility layer on top of dnf and since the
+        Ansible modules for yum(aka yum3) and dnf(aka yum4) call each of yum3
+        and yum4's python APIs natively on the backend, we need to handle this
+        here and pass off to the correct Ansible module to execute on the remote
+        system
+        '''
+
+        self._supports_check_mode = True
+        self._supports_async = True
+
+        result = super(ActionModule, self).run(tmp, task_vars)
+        del tmp  # tmp no longer has any effect
+
+        # Carry-over concept from the package action plugin
+        module = "auto"
+
+        try:
+            if self._task.delegate_to:  # if we delegate, we should use delegated host's facts
+                module = self._templar.template("{{hostvars['%s']['ansible_facts']['pkg_mgr']}}" % self._task.delegate_to)
+            else:
+                module = self._templar.template("{{ansible_facts.pkg_mgr}}")
+        except Exception:
```

Comment:
Maybe catch something more specific?

Reply:
I don't think the logic here is to catch something more specific.  But I always leave a comment in cases like these:
```
except Exception:
    # If module doesn't get set here, we fallback to running facts and trying again, below
    pass
```

Reply:
IMHO it's an antipattern when you catch everything wholesale. You mask future bugs this way.

Reply:
@webknjaz probably, I 100% just cribbed that from the package action plugin and didn't think much about the specific exception handler